### ISSUE TITLE
What is the behavior if connection/login status drops?

### ISSUE THREAD

**Issue by battaglia01** [kiwiz/gkeepapi#26]
The API documentation says to use login() to log in, then sync() periodically to make changes.

What happens if, for instance, the connection is dropped in between the login() and sync() calls? Will sync() automatically re-login, or does it silently fail?

If the latter, is there some way to check login() status to know if we need to login again?

The use case for this is, I'm trying to keep a Keep note synced with a KDE Plasma note. I have a script running that periodically checks, every 5 minutes or so, if the note has changed to sync the two. So my question is, should I just send a new login() request every 5 minutes? Or should I just login() once, then every five minutes call sync()? Or login() once, then every five minutes check login status and possibly re-login again, and then call sync()? etc.

I can't find anything on this in the documentation, so it might be good to add something there.

**Comment by kiwiz** [comment]
Tl;Dr There should be no need to call `login()` more than once. You can just call `sync()` repeatedly and that should work. More details below:

The code stores a `gpsoauth` "master token" after the initial login. This token allows us to mint OAuth tokens whenever we'd like (it doesn't expire). The `sync()` method checks if the token is valid and will automagically grab a new one before syncing.

**Comment by kiwiz** [comment]
You may want to catch `gkeepapi.exception.KeepException` in the case your code tries to call `sync()` when there is no internet connection.

**Comment by battaglia01** [comment]
OK thanks - works well. You might want to add that to the documentation but it seems to be working ok.
